### ISSUE TITLE
Float height on grid item

### ISSUE THREAD

**Issue by mathildefasst** [react-grid-layout/react-grid-layout#2098]
### Describe the bug

in your documentation, you indicate that the h property, of the grid item of the layout, accepts as type "number", which means that you are supposed to accept numbers with decimals (float) but also integers. However, when I have floating values, and I click on one of the items of the layout, it will reposition itself even though we have not dragged it. This problem does not occur if we have an integer as value. Here is a codesandbox to better understand: 

### Your Example Website or App

https://codesandbox.io/p/sandbox/q23kg7

### Steps to Reproduce the Bug or Issue

- create a project with the lib
- usage of GridLayout
- in the layout have at least 2 grid items with a floating value for the h property
- click on one of the items
- see that the item moves without dragging it

### Expected behavior

the item should not move if we did not drag it and just clicked on it

### react-grid-layout library version

1.4.1

### Operating System Version

macOs

### Browser

Safari

### Screenshots or Videos

https://github.com/user-attachments/assets/47f3538a-4c84-468b-9c71-28b8c248dcb4

**Comment by kateharrison990** [comment]
Yeah I encountered this same bug, did you find a fix for this @mathildefasst?

**Comment by mathildefasst** [comment]
@kateharrison990 I only fixed it by specifying integers instead of floats when rounding, but the original problem still exists unfortunately.

**Comment by STRML** [comment]
Layout item dimensions (w, h) are in grid units, not pixels, and should be integers. Float values can cause unexpected repositioning due to rounding during calculations.

**Solution:** Use integer values for h:
```jsx
// Instead of h: 2.5
{ i: 'a', x: 0, y: 0, w: 2, h: 3 }  // Use integers
```

If you need finer height control, use a smaller `rowHeight` value.
